packages/cli/src/ui/hooks/vim.ts

```diff
@@ -0,0 +1,1420 @@
+              Math.max(currentOffset, endOffset),
+              '',
+            );
+          }
+          break;
+        }
+
+        case CMD_TYPES.CHANGE_WORD_FORWARD: {
+          const text = buffer.text;
+          const currentOffset = getCurrentOffset();
+          let endOffset = currentOffset;
+          let offset = currentOffset;
+
+          for (let i = 0; i < count; i++) {
+            const nextWordOffset = findNextWordStart(text, offset);
+            if (nextWordOffset > offset) {
+              offset = nextWordOffset;
+            } else {
+              const wordEndOffset = findWordEnd(text, offset);
+              offset = Math.min(wordEndOffset + 1, text.length);
+              break;
+            }
+          }
+          endOffset = offset;
+
+          if (endOffset !== currentOffset) {
+            buffer.replaceRangeByOffset(
+              Math.min(currentOffset, endOffset),
+              Math.max(currentOffset, endOffset),
+              '',
+            );
+          }
+          dispatch({ type: 'SET_MODE', mode: 'INSERT' });
+          break;
+        }
+
+        case CMD_TYPES.CHANGE_WORD_BACKWARD: {
+          const text = buffer.text;
+          const currentOffset = getCurrentOffset();
+          let endOffset = currentOffset;
+          let offset = currentOffset;
+
+          for (let i = 0; i < count; i++) {
+            const prevWordOffset = findPrevWordStart(text, offset);
+            if (prevWordOffset < offset) {
+              offset = prevWordOffset;
+            } else {
+              break;
+            }
+          }
+          endOffset = offset;
+
+          if (endOffset !== currentOffset) {
+            buffer.replaceRangeByOffset(
+              Math.min(currentOffset, endOffset),
+              Math.max(currentOffset, endOffset),
+              '',
+            );
+          }
+          dispatch({ type: 'SET_MODE', mode: 'INSERT' });
+          break;
+        }
+
+        case CMD_TYPES.CHANGE_WORD_END: {
+          const text = buffer.text;
+          const currentOffset = getCurrentOffset();
+          let endOffset = currentOffset;
+          let searchOffset = currentOffset;
+
+          for (let i = 0; i < count; i++) {
+            const wordEndOffset = findWordEnd(text, searchOffset);
+            if (wordEndOffset <= searchOffset) break;
+
+            endOffset = wordEndOffset + 1;
+
+            if (i < count - 1) {
+              const nextWordStart = findNextWordStart(text, wordEndOffset + 1);
+              searchOffset = nextWordStart;
+              if (nextWordStart <= wordEndOffset) break;
+            }
+          }
+
+          if (endOffset > currentOffset) {
+            buffer.replaceRangeByOffset(currentOffset, endOffset, '');
+            dispatch({ type: 'SET_MODE', mode: 'INSERT' });
+          }
+          break;
+        }
+
+        case CMD_TYPES.DELETE_CHAR: {
+          for (let i = 0; i < count; i++) {
```

Comment:
needs to be moved to textBuffer

Reply:
this comment and the others in vim.ts along the same lines need to be addressed to make this code robust. this is the root cause of issues like the one you were seeing where pressing escape followed by `w` not behaving correctly. Right now due to this issue you could see `esc` followed by two `w` only behaving like `esc` followed by one `w`.
You can also reproduce this sort of issue by using ssh on a slow terminal.